audio-worklet/lib/param-controller.js

```diff
@@ -16,44 +16,54 @@
-   * @param  {String} options.type the type of input (only support for range)
-   * @param  {String} options.min the minimum possible value
-   * @param  {String} options.max the maximum possible value
-   * @param  {String} options.step the parameter's increment value
-   * @param  {String} options.default the default value of the parameter
+   * @param {String} parentId id for parent element.
+   * @param {Function} onChangeCallback Callback to trigger on input.
+   * @param {Object} options
+   * @param {String} options.name The name of the parameter.
+   * @param {String} options.id A variable name associated with the parameter.
+   * @param {String} options.type The type of input (only support for range).
+   * @param {Number} options.min The minimum possible value.
+   * @param {Number} options.max The maximum possible value.
+   * @param {Number} options.step The parameter's increment value.
+   * @param {Number} options.default The default value of the parameter.
    */
   constructor(parentId, onChangeCallback, options) {
     if (options == null) options = {};
+    
+    this.controller_ = document.createElement('input');
+    this.name_ = options.name || 'Parameter';
+    this.id_ = options.id || this.name_;
+    this.controller_.type = options.type || 'range';
+    this.controller_.min = options.min || 0;
+
+    if (options.max == null)
+      this.controller_.max = 1;
+    else
+      this.controller_.max = options.max;
+
+    if (options.step == null)
+      this.controller_.step = 0.1;
+    else
+      this.controller_.step = options.step;
+
+    this.controller_.step = options.step || 0;
+    this.controller_.value = options.default || 0;
+    this.onChangeCallback_ = onChangeCallback;
 
     // A param container holds controller and display.
     let container = document.createElement('div');
     document.getElementById(parentId).appendChild(container);
 
-    // Display div which updates as input is changed.
     this.header_ = document.createElement('div');
-    this.name_ = options.name || 'Parameter';
     this.header_.textContent = this.name_ + ': ';
-    this.header_.textContent += options.default || 1;
+    this.header_.textContent += this.controller_.value;
     container.appendChild(this.header_);
-
-    // Default to percentage scale for input unless specified in options.
-    this.controller_ = document.createElement('input');
-    this.controller_.type = options.type || 'range';
-    this.controller_.min = options.min || 0;
-    this.controller_.max = options.max || 100;
-    this.controller_.step = options.step || 1;
-    this.controller_.value = options.default || 1;
-    this.onChangeCallback_ = onChangeCallback;
-
-    if (this.controller_.type == 'range') {
+    
+    if (this.controller_.type == 'range')
       this.controller_.className += 'slider';
-    } else {
-      console.error(type + ' not defined');
-    }
-
+    else
+      throw (type + ' not defined');
```

Comment:
Wait. What is this `type`? Parentheses look redundant and please add a period at the end of sentence.